insitu/views/requirement.py

```diff
@@ -59,12 +77,29 @@ class RequirementListJson(ESDatatableView):
     permission_classes = (IsAuthenticated, )
 
 
-class RequirementAdd(ProtectedCreateView):
+class RequirementAdd(GetInitialMixing, ProtectedCreateView):
     template_name = 'requirement/add.html'
-    form_class = forms.RequirementForm
     model = models.Requirement
     permission_classes = (IsCopernicusServiceResponsible,)
 
+    def get_object(self):
+        pk = self.request.GET.get('pk', '')
+        if pk:
+            object = models.Requirement.objects.filter(pk=pk).first()
```

Comment:
try: return Requirement.objects.get except: notExists -> return None

Reply:
Move this to Mixin. If get_object -> return, otherwise, look for an object using GET pk